core/storage/user/gae_models.py

```diff
@@ -110,6 +110,22 @@ class UserSettingsModel(base_models.BaseModel):
     preferred_audio_language_code = ndb.StringProperty(
         default=None, choices=[
             language['id'] for language in constants.SUPPORTED_AUDIO_LANGUAGES])
+    # Displayed name for users on android. They are editable unlike usernames.
+    name = ndb.StringProperty(default=None)
+    # A code associated with profiles to provide a PIN based authentication
+    # within the account on android.
+    pin = ndb.StringProperty(default=None)
+    # List of profile user ids associated with a full user. None for profiles.
+    associated_profile_user_ids = ndb.StringProperty(repeated=True)
+    # Exploration language preferences specified by the user for android.
+    preferred_android_language_codes = ndb.StringProperty(
+        repeated=True,
+        indexed=True,
+        choices=[lc['code'] for lc in constants.SUPPORTED_CONTENT_LANGUAGES])
+    # Audio language preference used for audio translations on android.
+    preferred_android_audio_language_code = ndb.StringProperty(
+        default=None, choices=[
+            language['id'] for language in constants.SUPPORTED_AUDIO_LANGUAGES])
```

Comment:
Are these only related to profiles? I think it would be useful to groupt the fields by the category of use: both profiles and accounts, only profiles, only accounts. Or this info should be always mentioned in the comment.

Reply:
Grouped them now.